### PR TITLE
Model tools

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR brings `model-verifier` utility for checking the Spine type model at compile time.

To enable the check, add the following config to the build script:

```groovy
buildscript {
    dependencies {
        classpath "io.spine.tools:spine-model-verifier:${spineVersion}"
    }
}

apply plugin: 'io.spine.tools.spine-model-verifier'

compileJava {
    options.compilerArgs += ["-processor", "io.spine.model.AssignLookup", "-AspineDirRoot=${rootDir}"]
}

dependencies {
    compileOnly group: 'io.spine.tools', name: 'spine-model-assembler', version: spineVersion
}
```

_Note:_ in order for the utility to work properly, this config should be applied to all the Gradle projects of a single hierarchy.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.
Note: the changes shown as not covered by __codecov__ are covered by the integration tests.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

File: model/model-assembler/src/main/java/io/spine/model/assemble/SpineAnnotationProcessor.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+ * to the compiler.
+ *
+ * <p>Warning: do not use standard Java {@linkplain Throwable throwables} in the methods of this
+ * class to intentionally break the compilation. It may cause unreadable error messages. Use
+ * {@link #error(String) error()} method instead with the error message. Use throwables only
+ * to state the programming mistakes in the processor itself.
+ *
+ * @author Dmytro Dashenkov
+ */
+public abstract class SpineAnnotationProcessor extends AbstractProcessor {
```
Comment (armiol) [inline model/model-assembler/src/main/java/io/spine/model/assemble/SpineAnnotationProcessor.java]: It looks `@SPI`.

File: model/model-assembler/src/main/proto/spine/model/spine_model.proto [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// The message contains the type names of the Spine core entities found in a Spine project (not
+// necessarily grouped by the entity kind).
+//
+message SpineModel {
+
+    // All the command handling types of a Spine project.
+    //
+    // Each type is represented with its Java FQN.
+    //
+    // These types include `Aggregete`s, `ProcessManager`s and `CommandHandler`s.
```
Comment (armiol) [inline model/model-assembler/src/main/proto/spine/model/spine_model.proto]: Please find and fix a typo.

File: model/model-assembler/src/main/proto/spine/model/spine_model.proto [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+option java_package = "io.spine.model";
+
+// The Spine type model.
+//
+// This message represents raw (i.e. non-typed) model of a Spine project.
+//
+// The message contains the type names of the Spine core entities found in a Spine project (not
+// necessarily grouped by the entity kind).
+//
+message SpineModel {
```
Comment (armiol) [inline model/model-assembler/src/main/proto/spine/model/spine_model.proto]: Trying to look objectively it's really difficult to explain why this message is called the same as `io.spine.server.model.Model`, but has a weird structure.

That's why I would rename this message to `CommandHandlers`. And reduce the reserved field count to like `2 to 5`.

File: model/model-verifier/src/main/java/io/spine/model/verify/ModelVerifierPlugin.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.newInputStream;
+
+/**
+ * The plugin performing the Spine type model verification.
+ *
+ * @author Dmytro Dashenkov
+ */
+@Experimental
+public class ModelVerifierPlugin extends SpinePlugin {
```
Comment (armiol) [inline model/model-verifier/src/main/java/io/spine/model/verify/ModelVerifierPlugin.java]: Looks `final`.

File: model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/MalformedAggregate.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import io.spine.server.command.Assign;
+import io.spine.validate.AnyVBuilder;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+public class MalformedAggregate extends Aggregate<String, Any, AnyVBuilder> {
```
Comment (armiol) [inline model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/MalformedAggregate.java]: Requires some class-level comments.

File: model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/TestAggregate.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import io.spine.server.command.Assign;
+import io.spine.validate.AnyVBuilder;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+
+import java.util.Collections;
+import java.util.List;
+
+class TestAggregate extends Aggregate<String, Any, AnyVBuilder> {
```
Comment (armiol) [inline model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/TestAggregate.java]: Please have a valid class-level description and an `@author` tag. 

Also, if that's a `MailformedAggregate`'s vis-a-vis, I would name it accordingly.

File: model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/TestCommandHandler.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import io.spine.validate.AnyVBuilder;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+public class TestCommandHandler extends CommandHandler {
```
Comment (armiol) [inline model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/TestCommandHandler.java]: Same as above. Give these classes some love.

File: model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/TestProcMan.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import io.spine.validate.AnyVBuilder;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+public class TestProcMan extends ProcessManager<String, Any, AnyVBuilder> {
```
Comment (armiol) [inline model/model-verifier/src/test/resources/model-verifier-test/src/main/java/io/spine/model/verify/TestProcMan.java]: Same as above.

**Pull request merged**